src/OpenColorIO/ParseUtils.cpp

```diff
@@ -276,13 +277,14 @@ GpuLanguage GpuLanguageFromString(const char * s)
     const std::string str = StringUtils::Lower(p);
 
     if(str == "cg") return GPU_LANGUAGE_CG;
-    else if(str == "glsl_1.2") return GPU_LANGUAGE_GLSL_1_2;
-    else if(str == "glsl_1.3") return GPU_LANGUAGE_GLSL_1_3;
-    else if(str == "glsl_4.0") return GPU_LANGUAGE_GLSL_4_0;
+    else if(str == "glsl_1.2")    return GPU_LANGUAGE_GLSL_1_2;
+    else if(str == "glsl_1.3")    return GPU_LANGUAGE_GLSL_1_3;
+    else if(str == "glsl_4.0")    return GPU_LANGUAGE_GLSL_4_0;
     else if(str == "glsl_es_1.0") return GPU_LANGUAGE_GLSL_ES_1_0;
     else if(str == "glsl_es_3.0") return GPU_LANGUAGE_GLSL_ES_3_0;
-    else if(str == "hlsl_dx11") return GPU_LANGUAGE_HLSL_DX11;
-    else if(str == "osl_1") return LANGUAGE_OSL_1;
+    else if(str == "hlsl_dx11")   return GPU_LANGUAGE_HLSL_DX11;
+    else if(str == "osl_1")       return LANGUAGE_OSL_1;
+    else if(str == "metal")       return GPU_LANGUAGE_MSL_2_0;
```

Comment:
> Thanks for the work @Morteeza, I didn't found anything wrong with the code, only minor points. The style changes can also be addressed in a follow-up PR as you already mentioned. I hope we will be able to get rid of the manual declaration parsing code in the near future. Did you test the generated code in actual GPU processing code?

Hi @remia, 
Thanks a lot for your review. I applied the changes you suggested.
Previously we were not parsing the declaration part to extract variables, but since we wanted to keep this change as less intrusive as possible and not touch the interfaces defined in `OpenColorIO.h`, we decided to go this way. 

Also, yes. I have tried a few shaders generated with opencolorio changes and it seems to work fine. Just as an example, here you can see the input image is color adjusted with `spi-vfx` config from linear space to srgb8.

![in_Texture](https://user-images.githubusercontent.com/92918486/139601569-18377441-41ad-4ec0-b972-c28f5b1962ad.png)
![out_Texture](https://user-images.githubusercontent.com/92918486/139601575-fe1a07dc-6d8d-416a-8e82-94fdaa0b1ff9.png)

Reply:
> > @remia  The style changes can also be addressed in a follow-up PR as you already mentioned.

> @Morteeza Previously we were not parsing the declaration part to extract variables, but since we wanted to keep this change as less intrusive as possible and not touch the interfaces defined in OpenColorIO.h, we decided to go this way.

Yes. The idea was to try having the MSL support for the next patch release i.e. no ABI break. That's the only way I found for short-term but that's clearly not a long-term approach.